### DISCUSSION TITLE
What is the recommended setup for running vector.dev in a docker-compose?

### DISCUSSION THREAD

**Discussion by vedantroy** [vectordotdev/vector#18428] [Q&A]
I'm using docker-compose with a client + server container. What is the recommended setup for having vector.dev forward logs from both the client and server container to a sink?

**Comment by jszwedko** [comment]
Hi @vedantroy ,

The simplest way is probably to use the [Docker Logs](https://vector.dev/docs/reference/configuration/sources/docker_logs/) source. You will likely need to limit it to _just_ the other containers in the Docker Compose file. One of the `include_*` options on the source should help with tha (e.g. `include_containers` to specify a list of containers to tail).

**Reply by vedantroy** [reply]
Thank you for the response!

Hmm... when using `timberio/vector:0.10.0-alpine`, I'm getting:
```
hydrogen-vector-1      | Sep 18 05:30:56.465 ERROR source{name=docker type=docker}: vector::sources::docker: Listing currently running containers, faile
d error=Hyper error: hyper::Error(Connect, Compat { error: Os { code: 2, kind: NotFound, message: "No such file or directory" } })                   
hydrogen-vector-1      | Sep 18 05:30:56.465  INFO vector::shutdown: All sources have finished.                                                         
hydrogen-vector-1      | Sep 18 05:30:56.465  INFO vector: Shutting down.      
```

with this configuration:
```
# Sources
[sources.docker]
  type = "docker"
  include_containers = ["client2", "backend-ts"]

# New Sink for stdout
[sinks.console]
  type = "console"
  inputs = ["docker"]
  target = "stdout"
  encoding = "json"
```

Am I supposed to run vector outside of the docker-compose file and on the actual host machine?

**Reply by jszwedko** [reply]
Hey! For the `docker` source you need to mount the Docker socket into the container so that the source can access the Docker API. Also, do you mean to be running Vector 0.10.0? That version is a few years old.

**Reply by vedantroy** [reply]
Definitely not--will fix. And thanks for the tip about mounting, will try adding the following volume: `    /var/run/docker.sock:/var/run/docker.sock` and see if it works
